lib/screens/authentication/import_key/interactor/viewmodels/import_key_bloc.dart

```diff
@@ -32,8 +34,20 @@ class ImportKeyBloc extends Bloc<ImportKeyEvent, ImportKeyState> {
       emit(state.copyWith(pageState: PageState.failure, error: ImportKeyError.InvalidPrivateKey));
     } else {
       final results = await ImportKeyUseCase().run(publicKey);
-      emit(ImportKeyStateMapper()
-          .mapResultsToState(state, results, AuthDataModel.fromKeyAndWords(event.privateKey, event.words)));
+      final List<Result> alternateResults = event.alternatePrivateKey != null
+          ? await ImportKeyUseCase().run(CheckPrivateKeyUseCase().isKeyValid(event.alternatePrivateKey!)!)
+          : [];
```

Comment:
if privateKey is vali you fetch for alternativePrivateKey too.
Then you pass those 2 results lists to the mapper.

Reply:
I am pretty happy doing it there, seeing how well it works now - it took 1 second to find the passport account!

Reply:
The nature of these calls is this: 

Either the first one succeeds and the second fails, finding the LW key with the LW words
== OR ==
The first fails, and the second one succeeds, finding passport key
== OR ==
Both fail

It's impossible for the first and second call to both succeed.